### PR TITLE
New Common escape sequences cheat sheet

### PR DESCRIPTION
**Pull request by marioluigiman**
https://duck.co/ia/view/common_escape_sequences_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@marioluigiman Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub ID:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata post coming soon

File: share/goodie/cheat_sheets/json/commonescapesequence.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+  "id": "common_escape_sequences",
+  "name": "Common Escape Sequences",
+  "description": "Escape Sequences for common programming languages",
+
+  "metadata": {
+    "sourceName": "wikipedia",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: @marioluigiman Lets capitalise w

File: share/goodie/cheat_sheets/json/commonescapesequence.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{
+  "id": "common_escape_sequences",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: @marioluigiman  lets make this common_escape_sequences_cheat_sheet

**Comment by jophab** [comment]
@marioluigiman use hypens to seperate words in file name.

**Comment by jophab** [comment]
@marioluigiman Also I think there is little redundancy with content of name,description,and sectioname

**Comment by marioluigiman** [comment]
https://duck.co/ia/view/common_escape_sequences_cheat_sheet

**Comment by marioluigiman** [comment]
what else?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Common Escape Sequences guide)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/common_escape_sequences_mobile.png)

File: share/goodie/cheat_sheets/json/commonescapesequence.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        "val": "ASCII Bell (BEL)"
+      }, {
+        "key": "\\b",
+        "val": "ASCII Backspace (BS)"
+      }, {
+        "key": "\\f",
+        "val": "ASCII Formfeed (FF)"
+      }, {
+        "key": "\\n",
+        "val": "newline or ASCII Linefeed (LF)"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: let's reverse these so `ASCII` is art the beginning of the line, or at least capitalize `Newline` so it looks like the rest of the `val`

File: share/goodie/cheat_sheets/json/commonescapesequence.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  "template_type": "keyboard",
+
+  "section_order": ["Escape Sequences"],
+
+
+  "sections": {
+    "Escape Sequences": [
+      {
+        "val": "Backslash(\\)",
+        "key": "\\\\"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: Right now this produces a single backslash on the frontend, you need to double this.

---

I don't recall why I originally needed 4 backslashes to produce 1 on the frontend, so I will try and fix that later. For now, lets do `\\\\\\\\`

File: share/goodie/cheat_sheets/json/commonescapesequence.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
+{
+  "id": "common_escape_sequences_cheat_sheet",
+  "name": "Common Escape Sequences",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: I think "Escape Sequences" is a little vague. Perhaps "Character Escape Sequences" or "String Escape Sequences"?

@duckduckgo/community-leaders any thoughts?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=common escape sequences cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/common_escape_sequences_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/commonescapesequence.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  "description": "Escape Sequences for common programming languages",
+
+  "metadata": {
+    "sourceName": "Wikipedia",
+    "sourceUrl": "https://en.wikipedia.org/wiki/Escape_sequences_in_C"
+  },
+
+  "aliases": [
+    "common escape sequence", "common escape sequences", "common escape seq", "python escape sequences", "java escape sequences", "javascript escape sequences", "C escape sequences", "C++ escape sequences", "C# escape sequences"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: @marioluigiman Please avoid use of upper case letters in aliases

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=common escape sequences cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/common_escape_sequences_cheat_sheet_mobile.png)

**Comment by marioluigiman** [comment]
Fixed the \\ problem. Also made all aliases lowercase.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=common escape sequences cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/common_escape_sequences_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/commonescapesequence.json (hunk 5 above)
Comment (javathunderman) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: I suppose "Character Escape Sequences" would be good.

Comment (marioluigiman) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: Sure will do

Comment (javathunderman) [inline share/goodie/cheat_sheets/json/commonescapesequence.json]: @marioluigiman thanks!

**Comment by moollaza** [comment]
@marioluigiman apologies for the delay! this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@marioluigiman thanks again for creating the Common Escape Sequences Cheat Sheet! You did a great job. We at DuckDuckGo, the DuckDuckHack community, and all the people who will encounter this Instant Answer really appreciate it.

We could really use your help on something else now. We want to increase our Instant Answer coverage for as many programming languages and platforms as possible. We want a OpenGL Shading Language Cheat Sheet and I think you'd similarly do a great job with building that. Please update the [Programming IA Coverage table](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage) (mark it with a 'd') if you'd be willing to take it up!

I think it is a good fit for you, but I'm of course not entirely sure, so I'd love to chat if you're around this week or next. Please message me on slack (@moollaza) or send me an email (moollaza@duckduckgo.com) if you'd be interested in talking. 

Thanks again!

**Pull request merged**